edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using uPLibrary.Networking.M2Mqtt;
```

Comment:
We should hide the external dependency behind interfaces.

Reply:
I was considering it, but coming up a "general" library just to hide the concrete type did not seem worth it. First, the library has a quite narrow interface to use, second probably I could not create an interface that will fit exactly to the next library. So with the next library some changes need to be done anyway, and I thought it means less work to make the change in this class. (also meaning that the broker connector is a sort of abstraction toward the rest of the classes - supporting send/receive operations)